src/disk_io_thread.cpp

```diff
@@ -3179,10 +3144,14 @@ namespace libtorrent
 			add_completed_jobs(completed_jobs);
 	}
 
-	void disk_io_thread::thread_fun(int thread_id, thread_type_t type
-		, boost::shared_ptr<io_service::work> w)
+	void disk_io_thread::thread_fun(thread_type_t type
+		, std::shared_ptr<io_service::work> w)
 	{
-		DLOG("started disk thread %d\n", int(thread_id));
+		std::thread::id thread_id = std::this_thread::get_id();
+		std::stringstream thread_id_str;
+		thread_id_str << thread_id;
+
+		DLOG("started disk thread %s\n", thread_id_str.str().c_str());
```

Comment:
would you mind wrapping the stringstream and the loggin in #if DEBUG_DISK_THREAD?